src/cpp/shared/adventure/map.h

```diff
@@ -56,6 +56,41 @@ struct mapCellExtra {
 #endif
 };
 
+struct heroMapExtra {
+  char owner;
+  char couldBeHasArmy;
+  armyGroup army;
+  char field_11;
+  char heroID;
+  char artifacts[3];
+  char field_16;
+  int experience;
+  char couldBeHasSecondarySkills;
+  char secondarySkills[8];
```

Comment:
I'm going through your commits in order; I don't know if you already changed this.

Anyway, for constants like this that are meaningful to the gameplay and likely to show up elsewhere (i.e.: in the hero class), please use a named constant instead. I think the hero class may already have a NUM_SECONDARY_SKILLS constant.

On the other hand, that "char artifacts[3]" refers to how you can only set 3 artifacts for a hero in the map editor, which does not relate to anything other that heroes in the map editor, so it's okay to leave that one as the number 3.

Does this all make sense?

Reply:
Indeed.

Reply:
The NUM_SECONDARY_SKILLS constant doesn't exist. I'm going to define it in game.h